src/core/ddsc/tests/config.c

```diff
@@ -118,3 +118,423 @@ CU_Test(ddsc_config, incorrect_config, .init = ddsrt_init, .fini = ddsrt_fini) {
          "</"DDS_PROJECT_NAME">") == DDS_RETCODE_OK);
     CU_ASSERT_FATAL(dds_create_domain(2, "") == DDS_RETCODE_PRECONDITION_NOT_MET);
 }
+
+static bool patternmatch(const char* pattern, const char* string)
```

Comment:
The DDSI core has a function ``ddsi2_patmatch`` that does exactly this, I would be preferable to use that one instead. (Note: it'll have to be ``DDS_EXPORT``ed because of Windows ...)